File: src/test/java/org/micoli/php/DoctrineEntityListTest.java
package org.micoli.php;

import com.intellij.testFramework.fixtures.BasePlatformTestCase;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;
import java.util.Objects;
import java.util.stream.Collectors;
import org.micoli.php.configuration.ConfigurationException;
import org.micoli.php.configuration.ConfigurationFactory;
import org.micoli.php.configuration.exceptions.NoConfigurationFileException;
import org.micoli.php.symfony.list.DoctrineEntityElementDTO;
import org.micoli.php.symfony.list.DoctrineEntityService;
import org.micoli.php.symfony.list.configuration.DoctrineEntitiesConfiguration;

public class DoctrineEntityListTest extends BasePlatformTestCase {
    @Override
    protected String getTestDataPath() {
        return "src/test/resources/testData";
    }

    public void testItGetRoutesFromAttributes() {
        myFixture.copyDirectoryToProject("src", "/src");
        DoctrineEntityService DoctrineEntityListService = loadPluginConfiguration(getTestDataPath());
        List<DoctrineEntityElementDTO> lists = DoctrineEntityListService.getElements();
        String formattedList =
                lists.stream().map(DoctrineEntityElementDTO::name).sorted().collect(Collectors.joining(","));
        String expectedList = new ArrayList<>(Arrays.asList("article__article", "article__feed"))
                .stream().sorted().collect(Collectors.joining(","));
        assertEquals(expectedList, formattedList);
    }

    private DoctrineEntityService loadPluginConfiguration(String path) {
        DoctrineEntitiesConfiguration doctrineEntityListConfiguration = null;
        try {
            doctrineEntityListConfiguration = Objects.requireNonNull(ConfigurationFactory.loadConfiguration(path, 0L))
                    .configuration
                    .doctrineEntitiesConfiguration;
        } catch (ConfigurationException | NoConfigurationFileException e) {
            throw new RuntimeException(e);
        }
        DoctrineEntityService instance = DoctrineEntityService.getInstance(getProject());
        instance.loadConfiguration(myFixture.getProject(), doctrineEntityListConfiguration);

        return instance;
    }
}

File: src/test/kotlin/org/micoli/php/DoctrineEntityListTest.kt
package org.micoli.php

import com.intellij.testFramework.fixtures.BasePlatformTestCase
import org.assertj.core.api.Assertions.*
import org.micoli.php.configuration.ConfigurationException
import org.micoli.php.configuration.ConfigurationFactory
import org.micoli.php.configuration.exceptions.NoConfigurationFileException
import org.micoli.php.symfony.list.DoctrineEntityService

class DoctrineEntityListTest : BasePlatformTestCase() {
    override fun getTestDataPath(): String = "src/test/resources/symfony-demo"

    fun testItGetEntitiesFromAttributes() {
        myFixture.copyDirectoryToProject("src", "/src")
        val doctrineEntityListService = loadPluginConfiguration(testDataPath)
        val lists = doctrineEntityListService.getElements()
        val formattedList =
            lists.map { it.name }.filter { !it.contains("Proxies") }.sorted().joinToString(",")
        assertThat(formattedList)
            .isEqualTo("symfony_demo_comment,symfony_demo_post,symfony_demo_tag,symfony_demo_user")
    }

    private fun loadPluginConfiguration(path: String?): DoctrineEntityService {
        try {
            val doctrineEntityListConfiguration =
                ConfigurationFactory()
                    .loadConfiguration(path, 0L, true)
                    ?.configuration
                    ?.doctrineEntities
            val instance = DoctrineEntityService.getInstance(project)
            instance.loadConfiguration(doctrineEntityListConfiguration)

            return instance
        } catch (e: ConfigurationException) {
            throw RuntimeException(e)
        } catch (e: NoConfigurationFileException) {
            throw RuntimeException(e)
        }
    }
}
